packages/ethereum/contracts/src/HoprChannels.sol

```diff
@@ -331,25 +353,39 @@ contract HoprChannels is IERC777Recipient, ERC1820Implementer, Multicall {
   }
 
   /**
-   * @dev Finalize the channel closure, if cool-off period
-   * is over it will close the channel and transfer funds
-   * to the sender. Then emits {ChannelUpdated} and the
-   * {ChannelClosureFinalized} event.
+   * @dev Finalize the channel closure.
+   *
+   * For outgoing channel, if cool-off period is over it will close the channel
+   * and transfer funds to the sender.
+   * For incoming channel, close the channel immediately
+   *
+   * It emits {ChannelUpdated} and the {ChannelClosureFinalized} event.
+   * @param source the address of the source
    * @param destination the address of the counterparty
    */
-  function finalizeChannelClosure(address destination) external validateSourceAndDest(msg.sender, destination) {
-    (, Channel storage channel) = _getChannel(msg.sender, destination);
-    require(channel.status == ChannelStatus.PENDING_TO_CLOSE, 'channel must be pending to close');
-    require(channel.closureTime < _currentBlockTimestamp(), 'closureTime must be before now');
+  function finalizeChannelClosure(address source, address destination)
+    external
+    isSourceOrDest(msg.sender, source, destination)
+    validateSourceAndDest(source, destination)
+  {
+    (, Channel storage channel) = _getChannel(source, destination);
+    if (source == msg.sender) {
+      // source finalizes an outgoing channel closure
+      require(channel.status == ChannelStatus.PENDING_TO_CLOSE, 'channel must be pending to close');
+      require(channel.closureTime < _currentBlockTimestamp(), 'closureTime must be before now');
+    } else {
+      // destination finalize an incoming channel closure
+      require(channel.status != ChannelStatus.CLOSED, 'channel must not be already closed');
+    }
     uint256 amountToTransfer = channel.balance;
-    emit ChannelClosureFinalized(msg.sender, destination, channel.closureTime, channel.balance);
+    emit ChannelClosureFinalized(source, destination, channel.closureTime, channel.balance);
```

Comment:
The events are now missing the information who triggered the operation. This would still be useful.